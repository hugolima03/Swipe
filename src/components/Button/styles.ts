import { lighten } from 'polished'

import styled, { css, DefaultTheme } from 'styled-components'

type ButtonProps = {
  backgroundColor?: 'blue' | 'green'
  styleType: 'default' | 'outline' | 'minimal'
}

const backgrounds = {
  blue: (theme: DefaultTheme) => css`
    background-color: ${theme.colors.blue};
  `,
  green: (theme: DefaultTheme) => css`
    background-color: ${theme.colors.green};
  `
}

const buttonModifiers = {
  default: (theme: DefaultTheme) => css`
    background-color: ${theme.colors.primary};
  `,

  outline: (theme: DefaultTheme) => css`
    background-color: white;
    border: 1px solid ${theme.colors.primary};
    color: ${theme.colors.primary};

    &:hover {
      background-color: white;
    }
  `,
  minimal: (theme: DefaultTheme) => css`
    border: none;
    background-color: white;
    color: ${theme.colors.primary};
    border: 1px solid white;

    &:hover {
      background-color: white;
    }
  `
}

export const Button = styled.button<ButtonProps>`
  ${({ theme, backgroundColor, styleType }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    min-height: 3.6rem;

    font-family: ${theme.font.family};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};

    min-width: 6.7rem;
    border: none;
    color: white;
    border-radius: ${theme.border.radius};
    border: 1px solid ${theme.colors.primary};

    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.small};
    cursor: pointer;

    box-shadow: ${theme.shadows.default};

    transition: ${theme.transition.fast};

    position: relative;

    &:hover {
      background-color: ${lighten(0.1, theme.colors.primary)};
      border: 1px solid ${lighten(0.1, theme.colors.primary)};
    }

    &:disabled {
      outline: black;
      background-color: white;
      color: ${theme.colors.lightGray};
      border: 1px solid white;
      cursor: not-allowed;
    }

    ${backgroundColor && backgrounds[backgroundColor](theme)}

    ${buttonModifiers[styleType](theme)}
  `}
`
