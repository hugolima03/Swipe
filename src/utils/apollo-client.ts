import { ApolloClient, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: `${process.env.NEXT_PUBLIC_API_URL}`,
  cache: new InMemoryCache(),
  headers: { Authorization: `Bearer ${process.env.NEXT_PUBLIC_GRAPHCMS_TOKEN}` }
})

export default client
