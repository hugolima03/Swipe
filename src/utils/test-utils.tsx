import { MockedProvider, MockedResponse } from '@apollo/client/testing'

import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const mocks: MockedResponse<Record<string, unknown>>[] = []

const customRender = (
  ui: React.ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </MockedProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
