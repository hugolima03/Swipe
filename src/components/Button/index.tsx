import { ButtonHTMLAttributes } from 'react'

import * as S from './styles'

type ButtonTypes = ButtonHTMLAttributes<HTMLButtonElement>
// | AnchorHTMLAttributes<HTMLAnchorElement>

export type ButtonProps = {
  children: React.ReactNode
} & ButtonTypes

const Button = ({ children, ...props }: ButtonProps) => (
  <S.Button {...props}>{children}</S.Button>
)

export default Button
