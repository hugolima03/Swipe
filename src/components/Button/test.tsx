import { render } from 'utils/test-utils'
import Button from '.'

describe('<Button />', () => {
  it('should render the heading', () => {
    render(<Button>Teste</Button>)
  })
})
