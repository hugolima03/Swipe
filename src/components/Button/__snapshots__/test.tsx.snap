// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 3.6rem;
  font-family: Nunito Sans,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 0.8rem 1.6rem;
  min-width: 6.7rem;
  border: none;
  color: white;
  border-radius: 2.4rem;
  border: 1px solid #FF2773;
  font-weight: 700;
  font-size: 1.4rem;
  cursor: pointer;
  box-shadow: rgba(100,100,111,0.2) 0px 7px 29px 0px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  position: relative;
  background-color: #FF2773;
}

.c0:hover {
  background-color: #ff5a94;
  border: 1px solid #ff5a94;
}

.c0:disabled {
  outline: black;
  background-color: white;
  color: #B3ABBC;
  border: 1px solid white;
  cursor: not-allowed;
}

<button
  class="c0"
>
  Teste
</button>
`;
